### PR TITLE
IA-3859 Remove outdated Java version from unit tests instructions in readme

### PR DESCRIPTION
**Pull request by mlilynolting**
We upgraded past Java 8 in 2021 (commit 1b0cd116), so the line in the Readme below describes a requirement which no longer exists.

Also, editor removed some blank space at the ends of lines. (can revert those for a cleaner PR)

---
Have you read [CONTRIBUTING.md](https://github.com/DataBiosphere/leonardo/blob/develop/CONTRIBUTING.md) lately? If not, do that first.

I, the developer opening this PR, do solemnly pinky swear that:

- ~~[ ] I've documented my API changes in Swagger~~

In all cases:

- [x] Get a thumbsworth of review and PO signoff if necessary
- [x] Verify all tests go green (you can re-run automation tests with a comment saying `jenkins retest`
- [x] Run the automation tests multiple times in parallel to weed out instability if applicable via a comment saying `multi-test`
- [x] Squash and merge; Delete your branch after this
- [ ] Test this change deployed correctly and works on dev environment after deployment

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -100,8 +98,8 @@ To build jar, leonardo docker image, and leonardo-notebooks docker image
 ./docker/build.sh jar -d build
 ```
 
-To build jar, leonardo docker image, and leonardo-notebooks docker image 
-and push to repos `broadinstitute/leonardo` and `broadinstitute/leonardo-notebooks` 
+To build jar, leonardo docker image, and leonardo-notebooks docker image
```
Comment (Qi77Qi) [inline README.md]: `leonardo-notebooks` is no longer a thing either :) it's replaced with https://github.com/databiosphere/terra-docker

So we can just remove the mention of `leonardo-notebooks`

Comment (mlilynolting) [reply README.md]: Excellent! I will commit a change momentarily - please lmk if there ought to be a mention of https://github.com/databiosphere/terra-docker somewhere else.

Comment (mlilynolting) [reply README.md]: jenkins retest

**Review by Qi77Qi — APPROVED** [review]

**Comment by Qi77Qi** [comment]
woohoo...first leo PR!

**Comment by Qi77Qi** [comment]
jenkins retest

**Pull request merged**